src/basic/virt.c

```diff
@@ -227,15 +227,16 @@ static int detect_vm_xen(void) {
         return VIRTUALIZATION_XEN;
 }
 
-static bool detect_vm_xen_dom0(void) {
+/* Returns -errno, or 0 for domU, or 1 for dom0 */
+static int detect_vm_xen_dom0(void) {
         _cleanup_free_ char *domcap = NULL;
         char *cap, *i;
         int r;
 
         r = read_one_line_file("/proc/xen/capabilities", &domcap);
         if (r == -ENOENT) {
-                log_debug("Virtualization XEN not found, /proc/xen/capabilities does not exist");
-                return false;
+                log_debug("Virtualization XEN because /proc/xen/capabilities does not exist");
```

Comment:
was this mean to be "Virtualization XEN not found because /proc/xen/capabilities does not exist"?

Reply:
This is correct. At the point this code runs, it is already know that some "Xen domain" is active. Now it needs to be decided wether this is a "dom0" or a "domU". Without the capabilities file it is hard to tell, so XEN is assumed when xenfs is not mounted. 

There will be followup changes later today to fix #6662 for real. This change is just a warm-up.

Reply:
I think that @grooverdan meant that the sentence is grammatically incorrect, it seems to be missing a verb.

Reply:
Should it better read `"Virtualization XEN kept because /proc/xen/capabilities does not exist"`? Please suggest something if that would not be correct either.

Reply:
I think it's better (at such a low level) to just describe what happened, without trying to specify what the higher level logic will decide. Maybe say "/proc/xen/capabilities does not exist, assuming we're not not XEN dom0" ?